plugins/modules/ipaservice.py

```diff
@@ -217,20 +217,24 @@
     temp_kdestroy, valid_creds, api_connect, api_command, compare_args_ipa, \
     encode_certificate, gen_add_del_lists, module_params_get, to_text, \
     api_check_param
+import ipalib.errors
 
 
 def find_service(module, name):
     _args = {
         "all": True,
     }
 
-    _result = api_command(module, "service_find", to_text(name), _args)
-
-    if len(_result["result"]) > 1:
-        module.fail_json(
-            msg="There is more than one service '%s'" % (name))
-    elif len(_result["result"]) == 1:
-        _res = _result["result"][0]
+    try:
+        _result = api_command(module, "service_show", to_text(name), _args)
+    except ipalib.errors.NotFound as svcnf:
+        if "service not found" in str(svcnf):
```

Comment:
Why do we have to return None in one case and fail in the other? It would be nice to have a comment.

Reply:
This is an error. It was left from the use of the broader `Exception`.
Fixed.